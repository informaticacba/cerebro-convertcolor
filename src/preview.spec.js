import React from 'react';
import Preview from './preview'
import renderer from 'react-test-renderer';
import Color from 'color'

describe('preview', () => {
  const getColor = () => ({
    hex: '#000',
    rgb: 'rgb(0, 0, 0)',
    hsl: 'hsl(0, 0%, 0%)',
  })
  const getActions = () => ({
    copyToClipboard: jest.fn(),
    hideWindow: jest.fn()
  })
  
  it('renders preview and matches snapshot', () => {
    const component = renderer.create(
      <Preview color={getColor()} actions={getActions()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('choosing element copies color to clipboard', () => {
    const actions = getActions()
    const component = renderer.create(
      <Preview color={getColor()} actions={actions} />
    );
    const tree = component.toJSON();
    tree.children[0].children[0].props.onSelect()
    expect(actions.copyToClipboard).toBeCalledWith('#000')

    tree.children[0].children[1].props.onSelect()
    expect(actions.copyToClipboard).toBeCalledWith('rgb(0, 0, 0)')

    tree.children[0].children[2].props.onSelect()
    expect(actions.copyToClipboard).toBeCalledWith('hsl(0, 0%, 0%)')
  })
})