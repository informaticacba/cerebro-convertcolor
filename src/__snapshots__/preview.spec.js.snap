// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preview renders preview and matches snapshot 1`] = `
<div
  className={undefined}
>
  <div
    onKeyDown={[Function]}
  >
    <div
      className="undefined"
      onClick={[Function]}
      onKeyDown={[Function]}
      onSelect={[Function]}
      tabIndex={0}
    >
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "#000",
          }
        }
      />
      #000
    </div>
    <div
      className="undefined"
      onClick={[Function]}
      onKeyDown={[Function]}
      onSelect={[Function]}
      tabIndex={0}
    >
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "rgb(0, 0, 0)",
          }
        }
      />
      rgb(0, 0, 0)
    </div>
    <div
      className="undefined"
      onClick={[Function]}
      onKeyDown={[Function]}
      onSelect={[Function]}
      tabIndex={0}
    >
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "hsl(0, 0%, 0%)",
          }
        }
      />
      hsl(0, 0%, 0%)
    </div>
  </div>
</div>
`;
